src/libsyntax_ext/deriving/custom.rs

```diff
@@ -70,6 +70,11 @@ impl MultiItemModifier for ProcMacroDerive {
             }
         }
 
+        let item = item.map_attrs(|mut attrs| {
+            attrs.retain(|a| a.path != "structural_match" && a.path != "rustc_copy_clone_marker");
+            attrs
+        });
```

Comment:
@abonander 
Do you know why these attributes are filtered away here?
They are *normally* produced by builtin derives, but I'm not sure how they affect custom derives and why they need to be filtered away.

Reply:
@jseyfried added this in the original PR. Naive guess, it's avoiding exposing implementation details to the custom derive? Because these would be retained in the actual item since custom derives don't re-emit the item they're applied to.

Reply:
>custom derives don't re-emit the item they're applied to

Aha, this must be the key observation. Everything looks reasonable then.